test/integration/developer_portal/signup_test.rb

```diff
@@ -26,6 +26,48 @@ def test_create
     assert_response :redirect
   end
 
+  # A post request can't be sent directly to the controller bypassing the form
+  # when the spam protection is always enabled.
+  def test_create_spam_protected_always
+    @provider.settings.update(spam_protection_level: :captcha)
+    DeveloperPortal::SignupController.any_instance.stubs(:verify_captcha).returns(false)
+
+    post signup_path(account: {
+                       promo_code: '12345',
+                       org_name:   'alaska',
+                       user: {
+                         email:    'foo@example.edu',
+                         username: 'supertramp',
+                         password: 'westisthebest'
+                       }
+                     })
+
+    # The user wasn't created, so the spam protection worked
+    user = @provider.buyer_users.find_by(email: 'foo@example.edu')
+    assert_nil user
+  end
+
+  # A post request can't be sent directly to the controller bypassing the form
+  # when the spam protection is set to suspicious only.
+  def test_create_spam_protected_suspicious_only
+    @provider.settings.update(spam_protection_level: :auto)
+    DeveloperPortal::SignupController.any_instance.stubs(:verify_captcha).returns(false)
+
+    post signup_path(account: {
+                       promo_code: '12345',
+                       org_name:   'alaska',
+                       user: {
+                         email:    'foo@example.edu',
+                         username: 'supertramp',
+                         password: 'westisthebest'
+                       }
+                     })
```

Comment:
DIY. Consider extracting the account hash into a helper method or something.

Reply:
You meant `DRY`? :)
Although DIY also applies in this case :grimacing: 

I although thought about it, and suggested it [here](https://github.com/3scale/porta/pull/3109#discussion_r1036095882) but then I thought that it might actually reduce readability of the test. But I agree that it probably would be nice (rather than spending figuring our whether the request body is different in different tests)

Reply:
Done